source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
I disagree that it makes no sense to prescribe pressure.
If you have a velocity field that contains piecewise rigid body motions everywhere in the domain, the correct solution is that the (dynamic) pressure is zero everywhere in the domain. However, the finite element space cannot represent discontinuous velocity fields, so along the interface between rigid body regions (i.e. the subduction interface in this problem) there will be very large artificial pressure oscillations in order to satisfy momentum balance.